GCC 4.x fails to compile current develop

The failure originates in a failed cpantesters report http://www.cpantesters.org/cpan/report/4f5572f2-5d4d-11ef-a47d-351f6e8775ea (+many more similar)
```
cc -Iltm -Iltc/headers -DLTC_SOURCE -DLTC_NO_TEST -DLTC_NO_PROTOTYPES -DLTM_DESC -DHAS_FPSETMASK -DHAS_FLOATINGPOINT_H -fno-strict-aliasing -pipe -I/usr/local/include -DPIC -fPIC -O  -DARGTYPE=4 -c ltc/misc/pem/pem_pkcs.c -o ltc/misc/pem/pem_pkcs.o
ltc/misc/pem/pem_pkcs.c: In function 'pem_decode_pkcs_filehandle':
ltc/misc/pem/pem_pkcs.c:275: error: unknown field 'f' specified in initializer
ltc/misc/pem/pem_pkcs.c: In function 'pem_decode_pkcs':
ltc/misc/pem/pem_pkcs.c:287: error: unknown field 'buf' specified in initializer
ltc/misc/pem/pem_pkcs.c:287: warning: initialization from incompatible pointer type
ltc/misc/pem/pem_pkcs.c:287: error: unknown field 'buf' specified in initializer
ltc/misc/pem/pem_pkcs.c:287: error: unknown field 'buf' specified in initializer
ltc/misc/pem/pem_pkcs.c:287: warning: initialization makes integer from pointer without a cast
*** [ltc/misc/pem/pem_pkcs.o] Error code 1
```

Not sure if we want to support gcc-4.x.

--- a/src/headers/tomcrypt_private.h
+++ b/src/headers/tomcrypt_private.h
@@ -347,7 +347,7 @@ struct get_char {
       FILE *f;
 #endif /* LTC_NO_FILE */
       struct bufp buf;
-   };
+   } data;
    struct str unget_buf;
    char unget_buf_[LTC_PEM_DECODE_BUFSZ];
 };

--- a/src/misc/pem/pem_pkcs.c
+++ b/src/misc/pem/pem_pkcs.c
@@ -272,7 +272,7 @@ int pem_decode_pkcs_filehandle(FILE *f, ltc_pka_key *k, const password_ctx *pw_c
    LTC_ARGCHK(f != NULL);
    LTC_ARGCHK(k != NULL);
    {
-      struct get_char g = { .get = pem_get_char_from_file, .f = f };
+      struct get_char g = { .get = pem_get_char_from_file, .data.f = f };
       return s_decode(&g, k, pw_ctx);
    }
 }
@@ -284,7 +284,7 @@ int pem_decode_pkcs(const void *buf, unsigned long len, ltc_pka_key *k, const pa
    LTC_ARGCHK(len != 0);
    LTC_ARGCHK(k != NULL);
    {
-      struct get_char g = { .get = pem_get_char_from_buf, SET_BUFP(.buf, buf, len) };
+      struct get_char g = { .get = pem_get_char_from_buf, SET_BUFP(.data.buf, buf, len) };
       return s_decode(&g, k, pw_ctx);
    }
 }

--- a/src/misc/pem/pem_read.c
+++ b/src/misc/pem/pem_read.c
@@ -20,18 +20,18 @@ extern const unsigned long pem_dek_infos_num;
 #ifndef LTC_NO_FILE
 int pem_get_char_from_file(struct get_char *g)
 {
-   return getc(g->f);
+   return getc(g->data.f);
 }
 #endif /* LTC_NO_FILE */
 
 int pem_get_char_from_buf(struct get_char *g)
 {
    int ret;
-   if (g->buf.work == g->buf.end) {
+   if (g->data.buf.work == g->data.buf.end) {
       return -1;
    }
-   ret = *g->buf.work;
-   g->buf.work++;
+   ret = *g->data.buf.work;
+   g->data.buf.work++;
    return ret;
 }
 

--- a/src/misc/pem/pem_ssh.c
+++ b/src/misc/pem/pem_ssh.c
@@ -801,7 +801,7 @@ int pem_decode_openssh_filehandle(FILE *f, ltc_pka_key *k, const password_ctx *p
    LTC_ARGCHK(f != NULL);
    LTC_ARGCHK(k != NULL);
    {
-      struct get_char g = { .get = pem_get_char_from_file, .f = f };
+      struct get_char g = { .get = pem_get_char_from_file, .data.f = f };
       return s_decode_openssh(&g, k, pw_ctx);
    }
 }
@@ -839,7 +839,7 @@ int pem_decode_openssh(const void *buf, unsigned long len, ltc_pka_key *k, const
    LTC_ARGCHK(len != 0);
    LTC_ARGCHK(k != NULL);
    {
-      struct get_char g = { .get = pem_get_char_from_buf, SET_BUFP(.buf, buf, len) };
+      struct get_char g = { .get = pem_get_char_from_buf, SET_BUFP(.data.buf, buf, len) };
       return s_decode_openssh(&g, k, pw_ctx);
    }
 }

